diesel_tests/tests/transactions.rs

```diff
@@ -48,9 +40,9 @@ fn transaction_is_rolled_back_when_returned_an_error() {
     setup_test_table(&connection, test_name);
     let get_count = || count_test_table(&connection, test_name);
 
-    let _ = connection.transaction::<(), (), _>(|| {
+    let _ = connection.transaction::<(), _, _>(|| {
         connection.execute(&format!("INSERT INTO {} DEFAULT VALUES", test_name)).unwrap();
-        Err(())
+        Err(Error::__Nonexhaustive)
```

Comment:
It's super weird to use this variant. It's in a test though, so I can see why you did it instead of introducing an arbitrary new variant.